src/formatDate.js

```diff
@@ -8,7 +8,66 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYear = yearFormatAndPlace(fromFormat);
+  const newYear = yearFormatAndPlace(toFormat);
+
+  const formatedDate = [];
+  let year = '';
+
+  const dateParts = date.split(fromFormat[3]);
```

Comment:
There is no validation to check if the `date` string is in the correct format as specified by `fromFormat` before proceeding with the split operation. If the `date` string is not in the expected format, the function may not behave as intended.